airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -73,191 +73,129 @@
-                  syncWorkflowInputs.getDestinationLauncherConfig(),
-                  syncWorkflowInputs.getSyncInput(),
-                  connectionId));
-
-              final StandardSyncSummary standardSyncSummary = standardSyncOutput.get().getStandardSyncSummary();
-
-              if (workflowState.isResetConnection()) {
-                workflowState.setResetConnection(false);
-              }
-
-              if (standardSyncSummary != null && standardSyncSummary.getStatus() == ReplicationStatus.FAILED) {
-                failures.addAll(standardSyncOutput.get().getFailures());
-                partialSuccess = standardSyncSummary.getTotalStats().getRecordsCommitted() > 0;
-                workflowState.setFailed(true);
-              }
-            } catch (final ChildWorkflowFailure childWorkflowFailure) {
-              if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
-                // do nothing, cancellation handled by cancellationScope
-
-              } else if (childWorkflowFailure.getCause() instanceof ActivityFailure) {
-                final ActivityFailure af = (ActivityFailure) childWorkflowFailure.getCause();
-                failures.add(FailureHelper.failureReasonFromWorkflowAndActivity(
-                    childWorkflowFailure.getWorkflowType(),
-                    af.getActivityType(),
-                    af.getCause(),
-                    maybeJobId.get(),
-                    maybeAttemptId.get()));
-                throw childWorkflowFailure;
-              } else {
-                failures.add(
-                    FailureHelper.unknownOriginFailure(childWorkflowFailure.getCause(), maybeJobId.get(), maybeAttemptId.get()));
-                throw childWorkflowFailure;
-              }
-            }
-          }
-        });
-        syncWorkflowCancellationScope.run();
+        cancellableSyncWorkflow = generateSyncWorkflowRunnable(connectionUpdaterInput);
+        cancellableSyncWorkflow.run();
       } catch (final CanceledFailure cf) {
         // When a scope is cancelled temporal will thow a CanceledFailure as you can see here:
         // https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/workflow/CancellationScope.java#L72
         // The naming is very misleading, it is not a failure but the expected behavior...
       }
 
-      // The workflow state will be updated to true if a reset happened while a job was running.
-      // We need to propagate that to the new run that will be continued as new.
-      // When cancelling a reset, we endure that the next workflow won't be a reset.
-      // We are using a specific workflow state for that, this makes the set of the fact that we are going
-      // to continue as a reset testable.
+      if (workflowState.isDeleted()) {
+        deleteConnectionBeforeTerminatingTheWorkflow();
+        return;
+      }
+
       if (workflowState.isCancelledForReset()) {
```

Comment:
```suggestion
      // this means that the current workflow is being cancelled so that a reset can be run instead.
      if (workflowState.isCancelledForReset()) {
```

Reply:
Done